addons/sale_management/tests/test_sale_order.py

```diff
@@ -317,3 +317,24 @@ def test_option_creation(self):
             option.product_id = self.product_1
         order = order_form.save()
         self.assertTrue(bool(order.sale_order_option_ids.uom_id))
+
+    def test_option_price_unit_is_not_recomputed(self):
+        """
+        Verifies that user defined price unit for optional products remains the same after
+        update of quantities.
+        """
+
+        sale_order_with_option = self.env['sale.order'].create({
+            'partner_id': self.partner.id,
+            'sale_order_option_ids': [Command.create({
+                'product_id': self.optional_product.id,
+                'price_unit': 100,
+            })],
+        })
+
+        sale_order_with_option.sale_order_option_ids.price_unit = 10
```

Comment:
What i meant is that this line here could be integrated in the create above, to create the option with the price you want, AFAIK it doesn't bring anything else to modify the price directly after the line creation.

Reply:
got it :)